### PR TITLE
EN 6813 p2p debugger

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added p2p debugger component that will display bandwidth used by a node split by topic usage.
To enable it, run the node with `*:INFO,debug/p2p:TRACE`

### CODE REVIEW

File: debug/p2p/p2pDebugger.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/display"
+)
+
+var log = logger.GetOrCreate("debug/p2p")
+
+const printTimeOneSecond = time.Second //if this needs to be changed, remember to divide the values when computing metrics
```
Comment (sasurobert) [inline debug/p2p/p2pDebugger.go:17]: add a const for X * time.second - use X for division :)

File: debug/p2p/p2pDebugger.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	pd.cancelFunc = cancelFunc
+
+	go pd.doStats(ctx)
+
+	return pd
+}
+
+func (pd *p2pDebugger) checkLogTrace() bool {
```
Comment (sasurobert) [inline debug/p2p/p2pDebugger.go:69]: isLogTrace

File: debug/p2p/p2pDebugger.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+		fmt.Sprintf("%d / %s/s", m.outgoingRejectedNum, core.ConvertBytes(m.outgoingRejectedSize)),
+	}
+}
+
+type p2pDebugger struct {
+	selfPeerId          core.PeerID
+	mut                 sync.Mutex
+	data                map[string]*metric
+	cancelFunc          func()
+	shouldProcessDataFn func() bool
```
Comment (sasurobert) [inline debug/p2p/p2pDebugger.go:48]: no need for these 2 function pointers - shouldProcessDataFn and printStringFn

File: debug/p2p/p2pDebugger.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+		if !pd.shouldProcessDataFn() {
+			continue
+		}
+
+		str := pd.doStatsString()
+		pd.printStringFn(str)
+	}
+}
+
+func (pd *p2pDebugger) doStatsString() string {
```
Comment (sasurobert) [inline debug/p2p/p2pDebugger.go:142]: startsToString

File: debug/p2p/p2pDebugger.go [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+		m = &metric{
+			topic: topic,
+		}
+		pd.data[topic] = m
+	}
+
+	return m
+}
+
+func (pd *p2pDebugger) doStats(ctx context.Context) {
```
Comment (sasurobert) [inline debug/p2p/p2pDebugger.go:125]: maybe a better name ?

File: p2p/p2p.go [hunk 6]
```diff
@@ -277,3 +277,10 @@ type Cacher interface {
 	HasOrAdd(key []byte, value interface{}, sizeInBytes int) (ok, evicted bool)
 	IsInterfaceNil() bool
 }
+
+type Debugger interface {
```
Comment (sasurobert) [inline p2p/p2p.go:281]: comment to public interface

File: debug/p2p/p2pDebugger.go (hunk 1 above)
Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:17]: should have never wrote that comment :))))

Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:17]: done

File: debug/p2p/p2pDebugger.go (hunk 2 above)
Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:69]: done

File: debug/p2p/p2pDebugger.go (hunk 3 above)
Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:48]: I need them to properly test this struct

File: debug/p2p/p2pDebugger.go (hunk 4 above)
Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:142]: statsToString

File: debug/p2p/p2pDebugger.go (hunk 5 above)
Comment (iulianpascalau) [reply debug/p2p/p2pDebugger.go:125]: continuouslyPrintStatistics

**Pull request merged**